### PR TITLE
enable ecs collection in all e2e tests

### PR DESCRIPTION
**Pull request by kangyili**
What does this PR do?
---------------------

We’ve added ECS collection to the orchestration check, and we want to enable it for agents in the e2e tests. 

- https://github.com/DataDog/datadog-agent/pull/22060
- https://github.com/DataDog/datadog-agent/pull/21984

We want to broadly enable the collection internally before releasing it to the public.

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

File: components/datadog/agent/ecs.go [hunk 1]
```diff
@@ -144,6 +144,22 @@ func ecsLinuxAgentSingleContainerDefinition(e config.Env, apiKeySSMParamName pul
...
+				Name:  pulumi.StringPtr("DD_ORCHESTRATOR_EXPLORER_ECS_COLLECTION_ENABLED"),
+				Value: pulumi.StringPtr("true"),
+			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_CONTAINER_LIFECYCLE_ECS_TASK_EVENT_ENABLED"),
+				Value: pulumi.StringPtr("true"),
+			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_ECS_TASK_COLLECTION_ENABLED"),
+				Value: pulumi.StringPtr("true"),
```
Comment (kangyili) [inline components/datadog/agent/ecs.go:161]: Make workloadmeta use the latest v4 endpoint of ecs agent

File: components/datadog/agent/ecs.go [hunk 2]
```diff
@@ -144,6 +144,22 @@ func ecsLinuxAgentSingleContainerDefinition(e config.Env, apiKeySSMParamName pul
...
 			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_ORCHESTRATOR_EXPLORER_ENABLED"),
+				Value: pulumi.StringPtr("true"),
+			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_ORCHESTRATOR_EXPLORER_ECS_COLLECTION_ENABLED"),
+				Value: pulumi.StringPtr("true"),
+			},
+			ecs.TaskDefinitionKeyValuePairArgs{
```
Comment (kangyili) [inline components/datadog/agent/ecs.go:155]: Send ecs task lifecycle event

File: components/datadog/agent/ecs.go [hunk 3]
```diff
@@ -144,6 +144,22 @@ func ecsLinuxAgentSingleContainerDefinition(e config.Env, apiKeySSMParamName pul
 				Name:  pulumi.StringPtr("DD_TELEMETRY_CHECKS"),
 				Value: pulumi.StringPtr("*"),
 			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_ORCHESTRATOR_EXPLORER_ENABLED"),
+				Value: pulumi.StringPtr("true"),
+			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_ORCHESTRATOR_EXPLORER_ECS_COLLECTION_ENABLED"),
```
Comment (kangyili) [inline components/datadog/agent/ecs.go:152]: Send ecs task payload

File: components/datadog/agent/ecs.go [hunk 4]
```diff
@@ -144,6 +144,22 @@ func ecsLinuxAgentSingleContainerDefinition(e config.Env, apiKeySSMParamName pul
...
+				Value: pulumi.StringPtr("true"),
+			},
+			ecs.TaskDefinitionKeyValuePairArgs{
+				Name:  pulumi.StringPtr("DD_CONTAINER_LIFECYCLE_ECS_TASK_EVENT_ENABLED"),
```
Comment (kangyili) [inline components/datadog/agent/ecs.go:156]: Send ecs task lifecycle event

File: components/datadog/agent/ecs.go (hunk 1 above)
Comment (pducolin) [reply components/datadog/agent/ecs.go:149]: **❓ question**
Can you please double check with someone from containers this is ok with them?

**Review by pducolin — APPROVED** [review]

**Pull request merged**